validator/client/beacon-api/activation.go

```diff
@@ -0,0 +1,162 @@
+//go:build use_beacon_api
+// +build use_beacon_api
+
+package beacon_api
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/api/gateway/apimiddleware"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"google.golang.org/grpc"
+)
+
+func (c beaconApiValidatorClient) waitForActivation(ctx context.Context, in *ethpb.ValidatorActivationRequest) (ethpb.BeaconNodeValidator_WaitForActivationClient, error) {
+	// Represents the set of keys in `in`
+	targetPubKeys := make(map[string]struct{})
```

Comment:
Since you know the size of this map, help the runtime memory usage by specifying the exact size to pre-allocate.

```suggestion
	targetPubKeys := make(map[string]struct{}, len(in.PublicKeys))
```